### PR TITLE
Fix for #166 - `set_approval_for_all` fails if the caller owns a burnt token

### PR DESCRIPTION
**Pull request by davidatwhiletrue**

### CODE REVIEW

File: tests/src/burn.rs [hunk 1]
```diff
@@ -679,3 +679,94 @@ fn should_burn_token_in_hash_identifier_mode() {
...
+
+    let actual_operator: Option<Key> = support::call_entry_point_with_ret(
+        &mut builder,
+        *DEFAULT_ACCOUNT_ADDR,
+        nft_contract_key,
+        runtime_args! {
+            ARG_IS_HASH_IDENTIFIER_MODE => false,
+            ARG_TOKEN_ID => 1u64,
+        },
+        "get_approved_call.wasm",
```
Comment (gRoussac) [inline tests/src/burn.rs:762]: Hi @davidatwhiletrue, can you please replace all occurrences in this file of this wasm to `GET_APPROVED_WASM`

File: tests/src/burn.rs [hunk 2]
```diff
@@ -679,3 +679,94 @@ fn should_burn_token_in_hash_identifier_mode() {
...
+        "get_approved",
```
Comment (gRoussac) [inline tests/src/burn.rs:763]: Please add `get_approved` as entry point into constants and replace every occurrences thank you

File: contract/src/main.rs [hunk 3]
```diff
@@ -843,7 +843,7 @@ pub extern "C" fn set_approval_for_all() {
     for token_id in owned_tokens {
         // We assume a burned token cannot be approved
         if utils::is_token_burned(&token_id) {
```
Comment (gRoussac) [inline contract/src/main.rs:845]: > <s>This does not look legit to have a condition to only continue in all cases of the condition, these lines if not effective should be removed.</s>

<s>A also at the same time it seems strange to manipulate a burnt token. I'll check that with L2.</s>

**Review by gRoussac — APPROVED** [review]

**Comment by gRoussac** [comment]
@davidatwhiletrue, @darthsiroftardis I think the test checking the opposite can be then removed if ok with this change.

File: tests/src/burn.rs (hunk 1 above)
Comment (gRoussac) [reply tests/src/burn.rs:762]: I assume you will pull these changes from #168 thanks

Comment (davidatwhiletrue) [reply tests/src/burn.rs:762]: Once  #168 is merged I can rebase this and update.

**Comment by darthsiroftardis** [comment]
Closing in favor of #164

**Pull request closed without merging**